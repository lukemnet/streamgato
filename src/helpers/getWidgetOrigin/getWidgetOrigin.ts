import isItPreviewUrl from 'helpers/isItPreviewUrl/isItPreviewUrl';

const getWidgetOrigin = (origin: string) => {
  const siteUrl = window.location.origin;

  return isItPreviewUrl(siteUrl)
    ? siteUrl
    : origin;
};

export default getWidgetOrigin;
