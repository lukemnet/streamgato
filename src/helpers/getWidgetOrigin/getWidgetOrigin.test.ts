import getWidgetOrigin from './getWidgetOrigin';

const testPreviewOrigins = [
  'http://localhost:3000',
  'http://streamgato.lukem.net',
  'http://test.netlify.app'
];

const testNonPreviewOrigins = [
  'http://streamgato.com',
  'http://www.streamgato.com',
  'http://strmgt.com',
  'http://www.strmgt.com',
];

it('returns correct widget origin for a production URL', () => {
  delete window.location;

  testPreviewOrigins.forEach(origin => {
    window.location = {
      origin,
    } as any;

    expect(getWidgetOrigin(origin)).toMatchSnapshot();
  });
});

it('returns correct widget origin for a production URL', () => {
  delete window.location;

  testNonPreviewOrigins.forEach(origin => {
    window.location = {
      origin,
    } as any;

    expect(getWidgetOrigin(origin)).toMatchSnapshot();
  });
});
