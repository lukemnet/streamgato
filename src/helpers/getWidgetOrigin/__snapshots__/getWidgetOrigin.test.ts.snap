// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns correct widget origin for a production URL 1`] = `"http://localhost:3000"`;

exports[`returns correct widget origin for a production URL 2`] = `"http://streamgato.lukem.net"`;

exports[`returns correct widget origin for a production URL 3`] = `"http://test.netlify.app"`;

exports[`returns correct widget origin for a production URL 4`] = `"http://streamgato.com"`;

exports[`returns correct widget origin for a production URL 5`] = `"http://www.streamgato.com"`;

exports[`returns correct widget origin for a production URL 6`] = `"http://strmgt.com"`;

exports[`returns correct widget origin for a production URL 7`] = `"http://www.strmgt.com"`;
